src/Symfony/Component/Filesystem/Watcher/Resource/Locator/FileResourceLocator.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Filesystem\Watcher\Resource\Locator;
+
+use Symfony\Component\Filesystem\Watcher\Resource\ArrayResource;
+use Symfony\Component\Filesystem\Watcher\Resource\DirectoryResource;
+use Symfony\Component\Filesystem\Watcher\Resource\FileResource;
+use Symfony\Component\Filesystem\Watcher\Resource\ResourceInterface;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * @author Pierre du Plessis <pdples@gmail.com>
+ *
+ * @internal
+ */
+final class FileResourceLocator
+{
+    public function locate($path): ?ResourceInterface
+    {
+        if ($path instanceof Finder || $path instanceof \Iterator) {
+            $path = iterator_to_array($path);
+        }
+
+        if ($path instanceof \SplFileInfo) {
+            $path = $path->getRealPath();
+        }
+
+        if (\is_array($path)) {
+            return new ArrayResource(array_map([$this, 'locate'], $path));
+        }
+
+        if (\is_string($path)) {
+            if (is_dir($path)) {
+                return new DirectoryResource($path);
+            }
+
+            $paths = glob($path, \defined('GLOB_BRACE') ? GLOB_BRACE : 0);
+
+            if (1 === \count($paths)) {
```

Comment:
Why do we care here if array has single path and not let ArrayResource do it's job? ArrayResource surely works with single value array being passed in too?

Reply:
Small optimization. If we know there is only one file, then we don't need the loop in ArrayResource and can just check the file directly

Reply:
Right, but I would say super premature optimization. This isn't really worth it for increase of cyclomatic complexity.